dataservice_publisher/app.py

```diff
@@ -44,6 +51,140 @@ async def authenticated(request: web.Request) -> bool:
     return False
 
 
+class WeightedMediaRange:
+    """Class for handling weighted media ranges."""
+
+    def __init__(self, type: str, q: float = 1.0) -> None:
+        """Initialize the weighted media range."""
+        self.type = type.split("/")[0]
+        self.sub_type = type.split("/")[1]
+        self.q = q
+
+    def __eq__(self, other: Any) -> bool:  # pragma: no cover
+        """Compare two weighted media ranges."""
+        if isinstance(other, str):
+            return f"{self.type}/{self.sub_type}" == other
+        if isinstance(other, WeightedMediaRange):
+            return self.type == other.type and self.sub_type == other.sub_type
+        return False
+
+    def __str__(self) -> str:
+        """Return the weighted media range as a string."""
+        return f"{self.type}/{self.sub_type};q={self.q}"
+
+    def media_range(self) -> str:
+        """Return the media range."""
+        return f"{self.type}/{self.sub_type}"
+
+
+async def prepare_weighted_media_ranges(
+    accept_weighted_media_ranges: List[str],
+) -> List[str]:
+    """Prepare the accept weighted media ranges and sort on q-parameter."""
+    logging.debug(
+        f"Preparing accept weighted media ranges: {accept_weighted_media_ranges}"
+    )
+    # Assign q-parameter:
+    accept_weighted_media_ranges_sorted: List[WeightedMediaRange] = []
+
+    for accept_weighted_media_range in accept_weighted_media_ranges:
+        weighted_media_range_split = accept_weighted_media_range.split(";")
+        # Instantiate weighted media range:
+        try:
+            weighted_media_range = WeightedMediaRange(weighted_media_range_split[0])
+            logging.debug(
+                f"Assigning q-parameter for weighted media range: {accept_weighted_media_range}"
+            )
+            # If q-parameter is present, assign it:
+            for weighted_media_range_part in weighted_media_range_split[1:]:
+                if weighted_media_range_part.startswith("q="):
+                    weighted_media_range.q = float(
```

Comment:
Kan være greit å si noe om `0:5`
```suggestion
                    # RFC specifies only 3 decimals may be used in q value.
                    # Must strip additional decimals so that q bonus from specificity results in correct sorting.
                    weighted_media_range.q = float(
```

Reply:
Fiksa i cb0450451d891dad5df3f0f60339d3913b8f154b